src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/cardgrid/viewmodel/CardGridViewModel.kt

```diff
@@ -0,0 +1,103 @@
+package org.wycliffeassociates.otter.jvm.app.ui.cardgrid.viewmodel
+
+import com.github.thomasnield.rxkotlinfx.changes
+import com.github.thomasnield.rxkotlinfx.observeOnFx
+import com.github.thomasnield.rxkotlinfx.toObservable
+import io.reactivex.Observable
+import javafx.beans.property.SimpleBooleanProperty
+import javafx.collections.FXCollections
+import javafx.collections.ObservableList
+import org.wycliffeassociates.otter.common.data.model.ContentLabel
+import org.wycliffeassociates.otter.common.data.workbook.Chapter
+import org.wycliffeassociates.otter.common.data.workbook.Workbook
+import org.wycliffeassociates.otter.jvm.app.ui.cardgrid.CardData
+import org.wycliffeassociates.otter.jvm.app.ui.cardgrid.CardDataMapper
+import org.wycliffeassociates.otter.jvm.app.ui.workbook.viewmodel.WorkbookViewModel
+import org.wycliffeassociates.otter.jvm.utils.onChangeAndDoNow
+import tornadofx.*
+
+class CardGridViewModel: ViewModel() {
+    private val workbookViewModel: WorkbookViewModel by inject()
+
+    // List of content to display on the screen
+    // Boolean tracks whether the content has takes associated with it
+    val allContent: ObservableList<CardData>
+            = FXCollections.observableArrayList()
+    val filteredContent: ObservableList<CardData>
+            = FXCollections.observableArrayList()
+
+    // Whether the UI should show the plugin as active
+    private var showPluginActive: Boolean by property(false)
+    val showPluginActiveProperty = getProperty(CardGridViewModel::showPluginActive)
+
+    private var loading: Boolean by property(false)
+    val loadingProperty = getProperty(CardGridViewModel::loading)
+
+    val chapterModeEnabledProperty = SimpleBooleanProperty(false)
+
+    init {
+        Observable.merge(chapterModeEnabledProperty.toObservable(), allContent.changes()).subscribe { _ ->
+            filteredContent.setAll(
+                    if (chapterModeEnabledProperty.value == true) {
+                        allContent.filtered { it.item == ContentLabel.CHAPTER.value }
+                    } else {
+//                        allContent.filtered { it.item != ContentLabelEnum.CHAPTER.value }
+                        allContent
+                    }
+            )
+        }
+
+        workbookViewModel.activeWorkbookProperty.onChangeAndDoNow {
+            it?.let { wb -> loadChapters(wb) }
+        }
+
+        workbookViewModel.activeChapterProperty.onChange {
+            it?.let { ch -> loadChapterContents(ch) }
+                ?: workbookViewModel.activeWorkbookProperty.value
+                    ?.let { wb -> loadChapters(wb) }
+        }
```

Comment:
This is a little hard for me to read, and I think it's because you're using what appears to be a transformation chain but is actually a decision about which side-effect (Unit-return) function to call.  Maybe better to set this us with `if/else` or `when`.

Suggestion: Also, to my eyes, it's better to name the variable earlier than later, and use Kotlin's smart casts, instead of (e.g.) using `it` for a little while, then `ch` for the same value.

Reply:
I ended up just calling out the formatting but had the same feeling, and thought if/else would be better too

Reply:
Switched to `when`, it definitely is more readable.